scripts/generate_unknown_objects.py

```diff
@@ -0,0 +1,105 @@
+import argparse
+import logging
+from os.path import isdir
+from pathlib import Path
+from shutil import rmtree, copytree
+from typing import Mapping, Sequence
+
+import yaml
+
+situation_num: int = 0
+
+
+def get_situation_num(situation_dir: Path) -> int:
+    """Given a Path for a situation directory, return the associated number"""
+    return int(situation_dir.name.split('_')[-1])
+
+
+def copy_directory(start_path: Path, end_path: Path) -> None:
+    """Function to recursively copy directory from start_path to end_path"""
+    copytree(str(start_path), str(end_path))
+
+
+def copy_all_directories(output_dir: Path, dirs_to_copy: Sequence[Path]):
+    """Move all directories in dirs_to_copy to their corresponding location in the target directory"""
+    global situation_num
+    sorted_dirs = sorted(dirs_to_copy, key=get_situation_num)
+    for directory in sorted_dirs:
+        copy_directory(directory, output_dir / f'situation_{situation_num}')
+        situation_num += 1
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Utility to filter out specified objects from curriculum"
+    )
+    parser.add_argument('--input-dir', type=Path, help="Directory with curriculum samples")
+    parser.add_argument('--output-dir', type=Path, help="Directory to write remaining samples to")
+    parser.add_argument("--curriculum-name", required=False, type=str, help="Name of output curriculum")
+    parser.add_argument(
+        "-f", "--force",
+        action='store_true',
+        required=False,
+        help='Force overwrite of target directory. By default, the script exits with an error if there already exists '
+             'a directory at the target destination.'
+    )
+    parser.add_argument(
+        '-u', '--unknown-objects',
+        type=str,
+        nargs='+',
+        required=True,
+        help="List of unknown objects to be removed from curriculum"
+    )
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.INFO)
+
+    input_dir: Path = args.input_dir
+    output_dir: Path = args.output_dir
+    force_overwrite: bool = args.force
+    curriculum_name: str = args.curriculum_name
+    unknown_objects: Sequence[str] = args.unknown_objects
+    if not curriculum_name:
+        curriculum_name = output_dir.name
+
+    if not isdir(str(input_dir)):
+        logging.warning("Input directory does not exist")
+        raise FileExistsError(str(input_dir))
+
+    if isdir(str(output_dir)):
```

Comment:
Similar comment as above, although here maybe you do want to check if it's a directory based on the warning.

Reply:
Right, it's currently set up to create an output directory if one doesn't already exist, and it looks like `Path.is_dir()` is `False` if the directory doesn't exist. Also realized here I raised a `FileExistsError` instead of a `FileNotFoundError`, so I fixed that.